Description: Add parameter type checking on .bigstring
 If the parameter is not an integer, this would leave 65400 on the stack.
 Also, if there is no argument, signal stackunderflow.
 Seen with the PS sequence of bug 699794.
Origin: upstream, http://git.ghostscript.com/?p=ghostpdl.git;h=379be67
Author: Ray Johnston <ray.johnston@artifex.com>
Forwarded: yes
Bug: http://bugs.ghostscript.com/show_bug.cgi?id=699794
Last-Update: 2018-10-18
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Resource/Init/gs_ll3.ps
+++ b/Resource/Init/gs_ll3.ps
@@ -146,6 +146,10 @@
 } bind def
 
 /.bigstring {	% <size> .bigstring <string|bytestring>
+  count 1 lt { /.bigstring /stackunderflow signalerror } if
+  dup type /integertype ne {
+    /.bigstring /typecheck signalerror
+  } if
   dup 65400 gt { .bytestring } { string } ifelse
 } bind def
 
